src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -541,56 +541,63 @@ public Weight doCache(Weight weight, QueryCachingPolicy policy) {
             indexModule.addSearchOperationListener(new SearchOperationListener() {
 
                 @Override
-                public void onNewContext(SearchContext context) {
-
+                public void onPreQueryPhase(SearchContext context) {
                     if(advancedModulesEnabled) {
                         dlsFlsValve.handleSearchContext(context, threadPool, namedXContentRegistry);
                     }
                 }
 
                 @Override
-                public void onNewScrollContext(SearchContext context) {
-
-                    final ScrollContext scrollContext = context.scrollContext();
+                public void onNewReaderContext(ReaderContext readerContext) {
+                    final boolean interClusterRequest = HeaderHelper.isInterClusterRequest(threadPool.getThreadContext());
```

Comment:
nit: Would "bInterClusterRequest" be a better name given that "isInterClusterRequest" has been occupied by the method name?

Reply:
while I agree that `isInterClusterRequest` would be a better name it is better to keep the original name as the PR does not change how inter cluster request are processed.